docs/source/daml/reference/packages.rst

```diff
@@ -12,75 +12,94 @@ This page gives reference information on DAML package dependencies:
-If both projects ``foo`` as well as ``bar`` contain a module with the same name, the import can be
-disambiguated by adding the package name in front of the module name, e.g. ``import "bar"
-Some.Module``.
+.. code-block:: yaml
 
-Note that all modules of package ``foo`` that should be available as imports of other packages need
-to be exposed by adding them to the ``exposed-modules`` stanza of the `daml.yaml` file. If the
-``exposed-modules`` stanza is omitted, all modules of the project are exposed by default.
+    - /home/user/foo/foo.dar
 
-Importing archives compiled with different SDK's
-************************************************
+When you run ``daml build`` in ``bar`` project, the compiler will make the definitions in ``foo.dar`` available for importing. For example, if ``foo`` exports the module ``Foo``, you can import it in the usual way:
 
-All DAML archive dependencies of a project need to be compiled with the same SDK as the project
-itself. However, it is possible to import templates and data types of an archive compiled with an
-older SDK, by listing them under the ``data-dependencies`` stanza:
+.. code-block:: daml
 
-.. code-block:: yaml
+  import Foo
+
+Sometimes you will have multiple packages with the same exported module name. In that case, you must also specify which package the module comes from, as follows:
+
+.. code-block:: daml
+
+  import "foo" Foo
+
+By default, all modules of ``foo`` are made available when importing ``foo`` as a dependency. To control which modules of ``foo`` get exported, you may add an ``exposed-modules`` field in the ``daml.yaml`` file for ``foo``:
 
-  sdk-version: 0.0.0
-  name: foo
-  source: daml
-  version: 1.0.0
   exposed-modules:
-    - Some.Module
-    - Some.Other.Module
+  - Foo
+
+**Important Limitation:** For DAML archive imports via ``dependencies`` to work, the archive should be compiled with the same DAML SDK version. Otherwise, compilation will fail. If matching the SDK version is not possible, see the next section.
+
+Importing a DAML archive via data-dependencies
+**********************************************
+
+A secondary method for importing a DAML archive, which can be used when the DAML SDK versions do not match, is to import a DAML archive via the ``data-dependencies`` field in ``daml.yaml``:
+
+.. code-block:: yaml
+
   dependencies:
-    - daml-prim
-    - daml-stdlib
-    - /home/johndoe/bar/.daml/dist/bar.dar
+  - daml-prim
+  - daml-stdlib
   data-dependencies:
-    - /home/jondoe/bar-0.0.0/.daml/dist/bar-0.0.0.dar
+  - ../foo/foo.dar
 
-Modules from data dependencies can be imported as usual, but need to be qualified by the
-(generated) instances package:
+You can also import a ``.dalf`` file via data-dependencies.
 
-.. code-block:: daml
+When importing packages this way, the DAML compiler will attempt to reconstruct the DAML interface from the compiled DAML-LF binaries included in the DAML archive.
```

Comment:
This sounds a bit too much like it’s describing the implementation rather than what I care about as a user. Not quite sure what a better phrasing is.